poc-07.py

```diff
@@ -65,13 +65,36 @@ def search(self):
 
     def update(self, messages):
         """Update the storage.
-
         messages: is the collection of messages we have to create, update
         or remove."""
 
-        pass #TODO
-
-
+        """
+        Bug: If we provide try to update leftside first it will be updated as per right side changes provided by Engine. 
+             For Example: m2l is marked read and m2r is unread. Now when we are first trying to update left side 
+             with changes. We are making it unread on both the side.
+             Where as if we first update right side we will have m2l as well as m2r as read.
+             same goes for removal. Lets say there is m4l but m4r is missing . So what should we do should we remove m4l
+             or what.
+        """
+        for message in messages:
+            if message not in self.messages:
+                self.messages.append(message)  #Adding new message to the sel.messages storage
+            else: 
+                for s_message  in self.messages:            
```

Comment:
* Extra space before 'in'.
* Extra spaces the end of the line.
* "s_message": poor variable name.